cfgov/v1/__init__.py

```diff
@@ -39,38 +40,29 @@ def environment(**options):
         'choices_for_page_type': ref.choices_for_page_type,
         'is_blog': ref.is_blog,
         'get_page_state_url': share.get_page_state_url,
-        'parse_links': external_links,
+        'parse_links': external_links_filter,
     })
     env.filters.update({
         'slugify': slugify,
     })
     return env
 
-EXTERNAL_LINK_PATTERN = '(https?:\/\/(?:www\.)?(?![^\?]*(cfpb|consumerfinance).gov)(?!(content\.)?localhost).*)'
-
-
-def parse_link(soup):
-    try:
-        p = re.compile(EXTERNAL_LINK_PATTERN)
-
-        for a in soup('a'):
-            if p.match(a['href']):
-                a.append('<span class="' + str(
-                    os.environ.get('EXTERNAL_LINK_CSS',
-                                   'icon-link link-with-icon icon-link__external-link')) + '"></span>')
-    except:
-        pass
 
+def parse_links(soup):
+    p = re.compile(settings.EXTERNAL_LINK_PATTERN)
+    a_class = os.environ.get('EXTERNAL_A_CSS', 'icon-link icon-link__external-link')
+    span_class = os.environ.get('EXTERNAL_SPAN_CSS', 'icon-link_text')
+    for a in soup.find_all('a', href=True):
+        if p.match(a['href']):
```

Comment:
Can we update the href as well to point to the external redirect screen? This code, combined with the Wagtail redirect code, would allow use to delete the JS that is handling the redirects. It would also still work if JS is disabled?